src/sortStudents.ts

```diff
@@ -1,16 +1,67 @@
-
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grade',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const copyStudents: Student[] = [...students];
+
+  function calcAvg(student: Student): number {
+    const { grades } = student;
+
+    return grades.reduce((sum, grade) => sum + grade, 0) / grades.length;
+  }
+
+  if (sortBy) {
+    copyStudents.sort((stud1: Student, stud2: Student) => {
+      switch (sortBy) {
+        case SortType.Name:
+          return order !== 'desc'
+            ? stud1.name.localeCompare(stud2.name)
+            : stud2.name.localeCompare(stud1.name);
+
+        case SortType.Surname:
+          return order !== 'desc'
+            ? stud1.surname.localeCompare(stud2.surname)
+            : stud2.surname.localeCompare(stud1.surname);
```

Comment:
Checklist. `[CODE STYLE] - DRY. Combine logic for sorting by fields with same data types.` Same for other cases
```suggestion
    const sortOrder: number = order === 'asc' ? 1 : -1;
    copyStudents.sort((stud1: Student, stud2: Student) => {
      switch (sortBy) {
        case SortType.Name:
        case SortType.Surname:
          return sortOrder * stud1[sortBy].localeCompare(stud2[sortBy]);
```